napari/utils/_tests/test_progress.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+import pytest
+
+try:
+    import qtpy  # noqa
+except ImportError:
+    pytest.skip('Cannot test progress without qtpy.', allow_module_level=True)
+except RuntimeError:
+    pytest.skip(
+        'Cannot test progress without Qt bindings.', allow_module_level=True
+    )
+from ..progress import ProgressBar, progress
+
+
+def activity_dock_children(viewer):
+    return viewer.window.qt_viewer.activityDock.children()[4].children()
+
+
+def test_progress_with_iterable(make_napari_viewer):
+    viewer = make_napari_viewer()
+    r = range(100)
+    pbr = progress(r)
+
+    assert any(
+        isinstance(wdg, ProgressBar) for wdg in activity_dock_children(viewer)
+    )
```

Comment:
...then, using the context manager above: 

```suggestion
    with assert_pbar_added_to(viewer):
        r = range(100)
        pbr = progress(r)
```

(and similar for the tests below)